adapters/akka-http/src/main/scala/caliban/AkkaHttpAdapter.scala

```diff
@@ -32,17 +34,37 @@ class AkkaHttpAdapter private (private val options: AkkaHttpServerOptions)(impli
     requestInterceptor: RequestInterceptor[R] = RequestInterceptor.empty
   )(implicit
     runtime: Runtime[R],
+    materializer: Materializer,
     requestCodec: JsonCodec[GraphQLRequest],
-    responseCodec: JsonCodec[GraphQLResponse[E]]
+    responseCodec: JsonCodec[ResponseValue]
   ): Route = {
-    val endpoints = TapirAdapter.makeHttpService[R, E](
+    val endpoints: List[ServerEndpoint[ZioStreams, RIO[R, *]]] = TapirAdapter.makeHttpService[R, E](
       interpreter,
       skipValidation,
       enableIntrospection,
       queryExecution,
       requestInterceptor
     )
-    akkaInterpreter.toRoute(endpoints.map(TapirAdapter.convertHttpEndpointToFuture(_)))
+    akkaInterpreter.toRoute(
+      endpoints.map(endpoint =>
+        convertHttpStreamingEndpoint(
```

Comment:
Is there a performance degradation for using streaming in the case where we don't need it?

Reply:
Not necessarily applicable here since it all boils down to the underlying implementation, but streaming can incur a significant overhead. E.g., when Akka gRPC went from one of the lowest performing servers to the best performing one a year ago, it was mostly due to handling unary rpcs without streaming https://www.lightbend.com/blog/akka-grpc-update-delivers-1200-percent-performance-improvement.

I think as long as we don't stream the initial response we're probably safe, but might still be worth it to do a performance comparison just to understand it better. Happy to help with that if needed!

Reply:
I was thinking exactly about that article when I asked 😄

Reply:
Ok finally was able to get back to this, and make this change. By default now if there is no deferred block then it uses the fast single path.